specification/0041-hashed-mappings-specification.md

```diff
@@ -0,0 +1,146 @@
+but can always be creating from an existing official mapping set without knowledge of earlier hashed versions.
+This also means that it should be possible for anyone to generate hashed mappings,
+even if they don't plan to use the mappings-hasher designed by Quilt.
+In order to allow this, a specification for the hashed mapping is needed.
+
+In order to specify our intended behaviour, this document describes the process to generate a hashed mapping.
+Anything not specified in this RFC is considered a bug in the generation of hashed mappings.
+
+## Explanation
+
+Given an original mappings set and the corresponding classes, a new mapping set is created with the following properties:
+- All new mappings are a hash of the corresponding original mapping.
+- All new mappings are unique in the new mapping set where possible.
+- The new mappings must ensure that the resulting remapped jar can be verified and executed.
+- The new mapping set should be as resilient to changes in the original mapping set and corresponding classes as possible.
+
+In order to achieve this, the following operations are performed on the original mapping set:
+- Convert all class/method/field names into their corresponding "raw name".
+- Ensure methods in the same "override hierarchy" have the same raw name. (See "Method Name Sets".)
+- Hash all raw names using the same algorithm.
+- Mark all hashed class/method/field names with a corresponding prefix.
+- Prefix hashed class mappings with a default package.
+- Strip any mapping information that is unnecessary.
+
+### Raw Names
+For classes, the raw name is their binary name with the package omitted where possible.
+The package of a class can be omitted if and only if the original simple class name (i.e. the original class name without the package) is unique in the original mapping set.
+
+For methods and fields, the raw name is the raw name of the owner class, followed by their original name and descriptor, unless the descriptor can be omitted.
+The descriptor of a method/field can be omitted if and only if the original method/field name is unique in its owner class.
+
+The exact format is:
+- Class: `<class_name>`
+- Method: `m;<raw_class_name>.<original_method_name>;<original_descriptor>`
+- Field: `f;<raw_class_name>.<original_field_name>;<original_descriptor>`
+
+### Method Name Sets
+All methods in a name set are required to have the same raw name.
+Method A and method B are in the same name set if and only if one of the following is true:
+- A overrides B.
+- B overrides A.
+- There exists a method C that is both in the same name set as A and in the same name set as B.
+
+The raw name each set uses is the lexicographically lowest raw name of all methods in the name set that don't override another method.
+
+### Hash function
+The raw names are hashed using the SHA-256 hash algorithm.
+The resulting hash is converted into base-26 (a-z).
+The lowest 8 base-26 digits are then used as the hash.
```

Comment:
The order of the digits should be specified

Reply:
Is this good?